File: java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/DefaultPartition.java
package com.godaddy.asherah.appencryption;

public class DefaultPartition extends Partition {

  /**
   * Create a new {@code DefaultPartition} instance using the provided parameters.
   *
   * @param partitionId A unique identifier for a {@link Session}.
   * @param serviceId A unique identifier for a service, used to create a {@link SessionFactory} object.
   * @param productId A unique identifier for a product, used to create a {@link SessionFactory} object.
   */
  public DefaultPartition(final String partitionId, final String serviceId, final String productId) {
    super(partitionId, serviceId, productId);
  }

  @Override
  public String toString() {
    return getClass().getSimpleName() + "[partitionId=" + getPartitionId() +
      ", serviceId=" + getServiceId() + ", productId=" + getProductId() + "]";
  }
}

File: csharp/AppEncryption/AppEncryption/DefaultPartition.cs
namespace GoDaddy.Asherah.AppEncryption
{
  /// <summary>
  /// The default implementation of <see cref="Partition"/>.
  /// </summary>
  public class DefaultPartition : Partition
  {
    /// <summary>
    /// Initializes a new instance of the <see cref="DefaultPartition"/> class using the provided parameters.
    /// </summary>
    ///
    /// <param name="partitionId">A unique identifier for a <see cref="Session{TP,TD}"/>.</param>
    /// <param name="serviceId">A unique identifier for a service, used to create a <see cref="SessionFactory"/>
    /// object.</param>
    /// <param name="productId">A unique identifier for a product, used to create a <see cref="SessionFactory"/>
    /// object.</param>
    public DefaultPartition(string partitionId, string serviceId, string productId)
        : base(partitionId, serviceId, productId)
    {
    }

    /// <inheritdoc/>
    public override string ToString()
    {
      return GetType().Name + "[partitionId=" + PartitionId +
             ", serviceId=" + ServiceId + ", productId=" + ProductId + "]";
    }
  }
}
